packages/k2-alpine/src/components/Avatar/AvatarSelector.tsx

```diff
@@ -0,0 +1,93 @@
+import { Dimensions, ImageSourcePropType } from 'react-native'
+import React, { useMemo, useState } from 'react'
+import Carousel from 'react-native-reanimated-carousel'
+import { Pressable } from '../Primitives'
+import { Avatar } from './Avatar'
+
+const AvatarSelector = ({
+  avatars,
+  selectedId,
+  onSelect
+}: {
+  avatars: { id: string; source: ImageSourcePropType }[]
+  selectedId?: string
+  onSelect?: (id: string) => void
+}): JSX.Element => {
+  const data = useMemo(() => {
+    // we should always have an even number of avatars, due to infinite scrolling + two avatars per column
+    if (avatars.length % 2 === 0) {
+      return avatars
+    } else {
+      return [...avatars, ...avatars]
+    }
```

Comment:
I don’t want to restrict the number of avatars passed to `AvatarSelector` being even. This is because, in the future, when we allow users to use their NFTs, they will be combined with the preset avatars we provide, and the total number will be random.